oxo.py

```diff
@@ -1,33 +1,99 @@
+        strike_combinations -> Each position for strike
+    """
+    board = {}
+
+    board_width = 3
+    board_height = 3
+
+    strike_combinations = [
+        [(0, 0), (1, 0), (2, 0)],
+        [(0, 1), (1, 1), (2, 1)],
+        [(0, 2), (1, 2), (2, 2)],
+
+        [(0, 0), (0, 1), (0, 2)],
+        [(1, 0), (1, 1), (1, 2)],
+        [(2, 0), (2, 1), (2, 2)],
+
+        [(0, 0), (1, 1), (2, 2)],
+        [(2, 0), (1, 1), (0, 2)],
+    ]
+
+    def __init__(self, width = None, height = None, strikes = None):
+        # Allow the board to be edited on creation
+        self.board_width = width or self.board_width
+        self.board_height = height or self.board_height
+        self.strike_combinations = strikes or self.strike_combinations
+
+        # Move counter
+        self.total_moves = self.board_width * self.board_height
+        self.current_moves = 0
+
+        #Generate an empty board
+        for x in xrange(self.board_width):
+            for y in xrange(self.board_height):
+                self.board[x, y] = 0
 
     def get_square(self, x, y):
         """ Return 0, 1 or 2 depending on the contents of the specified square. """
-        raise NotImplementedError("TODO: implement get_square")
+        return self.board[x, y]
 
     def set_square(self, x, y, mark):
         """ If the specified square is currently empty (0), fill it with mark and return True.
             If the square is not empty, leave it as-is and return False. """
-        raise NotImplementedError("TODO: implement set_square")
+        if self.board[x, y] == 0:
+            self.board[x, y] = mark
+            self.current_moves += 1
+            return True
+        else:
+            return False
 
     def is_board_full(self):
         """ If there are still empty squares on the board, return False.
             If there are no empty squares, return True. """
-        raise NotImplementedError("TODO: implement is_board_full")
+        return self.current_moves >= self.total_moves
 
     def get_winner(self):
         """ If a player has three in a row, return 1 or 2 depending on which player.
             Otherwise, return 0. """
-        raise NotImplementedError("TODO: implement get_winner")
+        for line in self.strike_combinations: # For each strike combination
+            strike_search = self.board[line[0][0], line[0][1]] # The first position for the strike
+
+            # If the first position for the stike is 0, no point in checking the other positions
+            if strike_search == 0:
+                continue
+
+            # How many correct positions required in the current strike check to get a strike
+            parts_for_strike = len(line)
+            strike_complete = True
+
+            # For each position for strike line
+            for element in xrange(parts_for_strike):
+                x, y = line[element][0], line[element][1] # Get x and y from the current strike combination check
```

Comment:
This could have been written more succinctly (and more readably) as `for (x,y) in line:`